src/internal/direntry.rs

```diff
@@ -142,9 +142,10 @@ impl DirEntry {
         // Spec say this is suppose to be zero for DirEntries
         // but some cfb implementations set start_sector to FREE_SECTOR instead
         if obj_type == consts::OBJ_TYPE_STORAGE
-            && !(start_sector == 0 || start_sector == consts::FREE_SECTOR)
+            && !(start_sector == 0 || start_sector == consts::FREE_SECTOR
+            || start_sector == (consts::FREE_SECTOR - 1))
         {
-            malformed!("non-zero storage start sector: {}", start_sector);
+            malformed!("non-zero storage start sector: {:x} vs {:x}", start_sector, consts::FREE_SECTOR);
```

Comment:
Since zero is just one of several values we accept, "non-zero storage start sector" isn't really an accurate error anymore.  Maybe this line should be something like:

```rust
malformed!("invalid storage start sector: 0x{:X}", start_sector);
```